### DISCUSSION TITLE
Adding compression for dev server

### DISCUSSION THREAD

**Discussion by chneau** [farm-fe/farm#1516] [Show and tell]
Quick context:
I'm mainly coding from home, through my 5g connection (which might certainly be throttled because I use it a lot), through a SSH VSCode instance in Germany (NetCup love you!) from Scotland.

Moving from Vite to Farm was already a huge win both for build times and page loading in dev mode. 
With Vite I was having to load 700+ files and with my 5g connection and the latency (I think latency first?) it was taking a solid 15 seconds to land the page. Curse me if I created an error on the frontend that required me to F5 the page.

Not with farm it's down to 30 requests. Same kind of total size.

I was thinking now since the requests are less numerous but weightier, would makes even more sense for me to have compression on!

```ts
// bun install -d @types/koa @types/koa-compress koa-compress
// your farm.config.ts
const gzipPlugin: JsPlugin = {
	name: "gzip",
	configureDevServer(server) {
		server.app().middleware.unshift(compress()); // so that it's the first middleware and compress bundled js files and all!
	},
};
// and on your plugins down there
	plugins: [
		"@farmfe/plugin-react",
		"@farmfe/plugin-sass",
		gzipPlugin,
	],
```

From 15 mb to download down to 3 mb. that's me doing a hard refresh in 2.5 seconds.

Just wanted to share this and to say it has been a good experience to move from Vite to Farm.

And my next look at improving my own dev experience is around understanding why there is a `Keep-Alive: timeout=5` on the bundled js files and if I can change it. I'll play with it

**Comment by wre232114** [comment]
Thanks for the detailed context. It really inspires us! We'll add internal compression support later

**Comment by ErKeLost** [comment]
@chneau Thank you for using farm and your feedback. This is the best news I have heard today. I hope farm can help you save time. We will continue to work hard.
